schema/function_test.go

```diff
@@ -0,0 +1,517 @@
+package schema_test
+
+import (
+	"fmt"
+	"go.arcalot.io/assert"
+	"go.flow.arcalot.io/pluginsdk/schema"
+	"reflect"
+	"testing"
+)
+
+// Simple, no input or output.
+func TestCallableFunctionSchema_Simple(t *testing.T) {
+	called := false
+	simpleFunc, err := schema.NewCallableFunction(
+		"test",
+		make([]schema.Type, 0),
+		nil,
+		nil,
+		func() {
+			called = true
+		},
+	)
+	assert.NoError(t, err)
+	assert.Equals(t, called, false)
+	result, err := simpleFunc.Call([]any{})
+	assert.NoError(t, err)
+	assert.Nil(t, result)
+	assert.Equals(t, called, true)
+}
+
+// Simple, with one input.
+func TestCallableFunctionSchema_Simple1Param(t *testing.T) {
+	passedInVal := ""
+	simpleFunc, err := schema.NewCallableFunction(
+		"test",
+		[]schema.Type{schema.NewStringSchema(nil, nil, nil)},
+		nil,
+		nil,
+		func(test string) {
+			passedInVal = test
+		},
+	)
+	assert.NoError(t, err)
+	result, err := simpleFunc.Call([]any{"a"})
+	assert.NoError(t, err)
+	assert.Nil(t, result)
+	assert.Equals(t, passedInVal, "a")
+}
+
+// Simple, with no input, and only error out.
+func TestCallableFunctionSchema_SimpleWithNilErr(t *testing.T) {
+	simpleFunc, err := schema.NewCallableFunction(
+		"test",
+		make([]schema.Type, 0),
+		nil,
+		nil,
+		func() error {
+			return nil
+		},
+	)
+	assert.NoError(t, err)
+	result, err := simpleFunc.Call([]any{})
+	// No result or error
+	assert.NoError(t, err)
+	assert.Nil(t, result)
```

Comment:
So, I think the answer is that the `Call()` interface returns two values, a result and an error value:  the result will be the value returned by the called function if it returns one; if the function is not successfully invoked, or if the function returned no value, then the result will be `nil`.  Since, in this test scenario, the function does not return a value, the value of `result` should be `nil`, and it is appropriate for the test code to check that.